lib/include/domain_participant/builtin/discovery_config/DiscoveryConfig.hpp

```diff
@@ -0,0 +1,1057 @@
+ * @param[in] profile_id Domain participant profile identifier.
+ * @param[in] announcement_sec Builtin discovery lease announcement seconds.
+ * @param[in] announcement_nanosec Builtin discovery lease announcement nanoseconds.
+ *
+ * @throw ElementInvalid Exception if the provided lease announcement arguments are not valid.
+ */
+FASTDDS_QOS_PROFILES_MANAGER_DllAPI void set_lease_announcement(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& announcement_sec,
+        const std::string& announcement_nanosec);
+
+/**
+ * @brief Set the Domain Participant builtin discovery lease announcement (seconds).
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] profile_id Domain participant profile identifier.
+ * @param[in] announcement_sec Builtin discovery lease announcement seconds.
+ *
+ * @throw ElementInvalid Exception if the provided seconds are not valid.
+ */
+FASTDDS_QOS_PROFILES_MANAGER_DllAPI void set_lease_announcement_sec(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& announcement_sec);
+
+/**
+ * @brief Set the Domain Participant builtin discovery lease announcement (nanoseconds).
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] profile_id Domain participant profile identifier.
+ * @param[in] announcement_nanosec Builtin discovery lease announcement nanoseconds.
+ *
+ * @throw ElementInvalid Exception if the provided nanoseconds are not valid.
+ */
+FASTDDS_QOS_PROFILES_MANAGER_DllAPI void set_lease_announcement_nanosec(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& announcement_nanosec);
+
+/**
+ * @brief Set the Domain Participant builtin discovery number of initial announcements.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] profile_id Domain participant profile identifier.
+ * @param[in] count Builtin discovery number of initial announcements.
+ *
+ * @throw ElementInvalid Exception if the provided number of initial announcements is not valid.
+ */
+FASTDDS_QOS_PROFILES_MANAGER_DllAPI void set_initial_announcements_count(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& count);
+
+/**
+ * @brief Set the Domain Participant builtin discovery initial announcements duration.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] profile_id Domain participant profile identifier.
+ * @param[in] period_sec Builtin discovery initial announcements duration seconds.
+ * @param[in] period_nanosec Builtin discovery initial announcements duration nanoseconds.
+ *
+ * @throw ElementInvalid Exception if the provided initial announcements duration arguments are not valid.
+ */
+FASTDDS_QOS_PROFILES_MANAGER_DllAPI void set_initial_announcements_period(
```

Comment:
This `set` function with both `sec` and `nanosec` arguments is no longer required, we changed it into two individual set functions for `sec` and `nanosec`